File: androidApp/src/main/java/com/niranjan/khatri/time/android/PlatformContentView.kt
package com.niranjan.khatri.time.android

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material.Divider
import androidx.compose.material.MaterialTheme
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.niranjan.khatri.time.Platform
import kotlin.math.max
import kotlin.math.min

@Composable
fun PlatformContentView(){
    val items = platformItems()
    LazyColumn( modifier = Modifier.fillMaxSize()){
        items(items){ row ->
            RowView(title = row.first, desc = row.second )
        }
    }
}

@Composable
private fun RowView(
    title: String,
    desc: String
){
    Column(modifier = Modifier.fillMaxWidth()) {
        Column(Modifier.padding(8.dp)) {
            Text(text = title, color = Color.Blue, style = MaterialTheme.typography.caption)
            Text(text = desc, style = MaterialTheme.typography.body1)
        }
        Divider()
    }
}

private fun platformItems() : List<Pair<String, String>>{
    val platform = Platform()
    val items = mutableListOf(
        Pair("OS Name:", "${platform.operatingSystemName}"),
        Pair("OS Version:", "${platform.operatingSystemVersion}"),
        Pair("Cpu Type:", "${platform.cpuType}"),
        Pair("Device Model:", "${platform.deviceModel}")
    )

    platform.screenInfo?.let {
        val max = max(it.width, it.height)
        val min = min(it.width, it.height)
        items.add(Pair("Display", "${max}x ${min} @${it.density}"))
    }

    return items
}

File: iosApp/iosApp/PlatformContentView.swift
//
//  PlatformContentView.swift
//  iosApp
//
//  Created by Khatri, Niranjan on 03/09/23.
//  Copyright © 2023 orgName. All rights reserved.
//

import SwiftUI
import shared


private struct RowItem : Hashable {
    let title: String
    let desc: String
}

/*
 private let platformItems: [RowItem] = {
 let platform = Platform()
 
 var result: [RowItem] = [
 .init(
 title: "OS Name: ", desc: platform.operatingSystemName),
 .init(
 title: "OS Version: ",
 desc: platform.operatingSystemVersion
 ),
 .init(
 title: "Cpu Type: ",
 desc: platform.cpuType
 ),
 .init(
 title: "Device Model: ",
 desc: platform.deviceModel
 ),
 ]
 
 if let screen = platform.screenInfo {
 let width = min(screen.width, screen.height)
 let height = max(screen.width, screen.height)
 result.append(
 .init(title: "Display", desc: "\(width)x\(height) @\(screen.density)" )
 )
 }
 return result
 }()
 
 struct PlatformContentView: View {
 var body: some View {
 List{
 ForEach(platformItems, id: \.self){ item in
 VStack(alignment: .leading){
 Text(item.title).font(.footnote).foregroundColor(.secondary)
 Text(item.desc).font(.body).foregroundColor(.primary)
 }.padding(.vertical, 4)
 }
 }
 }
 }
 
 struct PlatformContentView_Previews: PreviewProvider {
 static var previews: some View {
 PlatformContentView()
 }
 }
 */
